File: csharp/armstrong-numbers/ArmstrongNumbers.cs
using System;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number)
    {
        string s = number.ToString();
        int retVal = 0;
        for (int i =0; i < s.Length; i++) 
        {
            retVal += (int)Math.Pow(Char.GetNumericValue(s[i]), s.Length);
        }
        return retVal == number;
    }
}

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

	boolean isArmstrongNumber(int numberToCheck) {

		String s = Integer.toString(numberToCheck);
		int tmpNum = 0;
		for (int i=0; i< s.length(); i++) {
			tmpNum += (int)Math.pow((s.charAt(i))-'0', s.length());
		}
		return (tmpNum == numberToCheck);
	}

}
